include/nnstreamer-edge.h

```diff
@@ -96,119 +96,362 @@ typedef void (*nns_edge_data_destroy_cb) (void *data);
+ * ret = nns_edge_publish (edge_h, data_h);
+ *
+ * // Destroy data handle.
+ * ret = nns_edge_data_destroy (data_h);
+ *
+ * // [Optional] Disconnect from the query-server or publisher. Or the connection is closed when edge handle is released.
+ * nns_edge_disconnect (edge_h);
  */
 int nns_edge_connect (nns_edge_h edge_h, const char *dest_host, int dest_port);
 
 /**
  * @brief Disconnect from the destination node.
+ * @param[in] edge_h The edge handle.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
  */
 int nns_edge_disconnect (nns_edge_h edge_h);
 
 /**
  * @brief Publish a message to desination (broker or connected node).
+ * @param[in] edge_h The edge handle.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ * @retval #NNS_EDGE_ERROR_IO Failed to transfer the data.
  */
 int nns_edge_publish (nns_edge_h edge_h, nns_edge_data_h data_h);
 
 /**
  * @brief Subscribe a message from broker.
+ * @param[in] edge_h The edge handle.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
  */
 int nns_edge_subscribe (nns_edge_h edge_h);
 
 /**
  * @brief Unsubscribe a message.
+ * @param[in] edge_h The edge handle.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
  */
 int nns_edge_unsubscribe (nns_edge_h edge_h);
 
 /**
  * @brief Set nnstreamer edge info.
  * @note The param key is case-insensitive. If same key string already exists, it will replace the old value.
+ * @param[in] edge_h The edge handle.
+ * @param[in] key Identifiers to determine which value to set.
+ * @param[in] value The value to be set.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ * @section table_sec Edge info table
+ * key                  | value
+ * ---------------------|--------------------------------------------------------------
+ * CAPS or CAPABILITY   | capapility strings.
+ * IP or HOST           | IP address of the node to accept connection from other node.
+ * PORT                 | Port of the node to accept connection from other node.
+ * DEST_IP or DEST_HOST | IP address of the destination node.
```

Comment:
describe the usage for the case we have broker (MQTT) or broker-discovery (AITT).